brainzutils/musicbrainz_db/release_group.py

```diff
@@ -136,6 +136,7 @@ def get_release_groups_for_artist(artist_id, release_types=None, limit=None, off
     if release_types is None:
         release_types = []
     release_types = [release_type.capitalize() for release_type in release_types]
+    release_types = ['EP' if release_type == 'Ep' else release_type for release_type in release_types]
```

Comment:
Can you explain how this will fix the underlying problem? I think that this method has major problems because it has no documentation or verification of what the release_types argument should be.
What stops me from entering "ep" as an argument here? That won't be caught by this check. You've also removed the `.capitalize()` call that was previously here. Will that break any existing code?
The musicbrainz documentation says that these items are in lower-case: https://musicbrainz.org/doc/MusicBrainz_API#Release_.28Group.29_Type_and_Status

I would like this method to be updated to include documentation that says what these values should be, either listed here, or as a link to the musicbrainz documentation, and something that describes what capitalisation they should have. The tests should be updated to show that different upper/lower-case versions work as expected.

Alternatively, does it make sense to have constants here, or something that make it more explicit what kind of values are expected?

Reply:
This is the output I got when I fetched the primary types from my `musicbrainz_db` instance.
```
SELECT * FROM musicbrainz.release_group_primary_type;
 id |   name    | parent | child_order | description |                 gid                  
----+-----------+--------+-------------+-------------+--------------------------------------
  1 | Album     |        |           1 |             | f529b476-6e62-324f-b0aa-1f3e33d313fc
  2 | Single    |        |           2 |             | d6038452-8ee0-3f68-affc-2de9a1ede0b9
  3 | EP        |        |           3 |             | 6d0c5bf6-7a33-3420-a519-44fc63eedebf
 12 | Broadcast |        |           4 |             | 3b2e49e1-2875-37b8-9fa9-1f7cf3f49900
 11 | Other     |        |          99 |             | 4fc3be2b-de1e-396b-a933-beb8f1607a22
```

I didn't remove `.capitalize()` for the reasons you already mentioned. After `.capitalize()` has done its work, the `release_types` is further modified only to replace `Ep` with `EP`.  The `.capitalize()` call before at Line 138 ensure `ep` gets converted to `Ep`. Existing code should not break because `ER`s never worked in the first place and other types are kept the same.

Regarding the documentation in musicbrainz, this is indeed weird because the MusicBrainz Web Service rejects types other than lower case ones.

https://musicbrainz.org/ws/2/release-group?artist=fc817311-4f42-4985-bc42-78b671d1dc18&type=Album&fmt=json
```
{
error: "Album is not a recognized release-group type.",
help: "For usage, please see: https://musicbrainz.org/development/mmd"
}
```
This https://musicbrainz.org/ws/2/release-group?artist=fc817311-4f42-4985-bc42-78b671d1dc18&type=album&fmt=json works as expected. 

Maybe someone from the MB team can shed some light on this.

Using constants and documenting the method is indeed better. I'll work on it.

Reply:
reo confirmed that "most of stuff is named capitalized".

Reply:
> I didn't remove .capitalize() for the reasons you already mentioned. 

Ah, I didn't notice that. Good catch.

I think that clearer documentation and/or constants definitely is the correct way to go. We could even do something like

```py
mapping = {'album': 'Album',
      'single': 'Single',
       'ep': 'EP,
       ... }
for each param in arguments:
    if param not in mapping.keys():
        raise...
    else:
        actual_arguments.append(mapping[param])
```
or something. This is a bit more code, but it's significantly more explicit in the code to see exactly what's happening.

Let me know if you come up with a good constants solution. Ideally we would use something like python 3 enums. We still need py 2 support in this project for acousticbrainz, but at the moment AB isn't using this file. I would be open to making this part of the library py3-only.

Reply:
The only python2 app that we have that seriously uses BU is AB, and that's going to be upgraded soon. I really don't mind using some python-3 only code here if needed.

I was also going to comment that we should look into _how_ this code is used. What projects use this method? Is it called with user-provided arguments, or are they always hard-coded in the calling code? Do they come from the output of another query? This will also affect the choice of which method to use.

Reply:
Let's get this merged. I would at the very least like a comment here that explains why we special-case the EP value here.